pkg/collector/corechecks/snmp/checkconfig/config.go

```diff
@@ -369,6 +372,16 @@ func NewCheckConfig(rawInstance integration.Data, rawInitConfig integration.Data
 	}
 	c.BulkMaxRepetitions = uint32(bulkMaxRepetitions)
 
+	if instance.Namespace != "" {
+		c.Namespace = instance.Namespace
+	} else {
```

Comment:
why no `initConfig.Namespace` ?

Reply:
Initially, I added `instances[].namespace` config mainly for SNMP Listener to be able to pass `namespace` config to the integration.

I didn't added `init_config.namespace` since I'm not sure yet which use case will require it. If user want to set `namespace` to all instances, they can already do it in `datadog.yaml` as `network_devices.namespace`.

`init_config.namespace` can be useful **if** user have multiple SNMP conf.yaml files and want to set a different namespace for all instances of each file.